os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -259,8 +279,12 @@ int se_ameba_hal_remove_key(hal_key_type mode, uint32_t key_idx)
 	AWRAP_ENTER;
 	int ret;
 
-	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
-		return HAL_INVALID_SLOT_RANGE;
+	ret = se_ameba_index_chk(key_idx);
+	if (HAL_SUCCESS != ret) {
+		return ret;
+	}
+	if (key_idx < FACTORY_KEY_INDEX_MAX) {	/* Factory Key Slot */
```

Comment:
same here

Reply:
All key index check will be done in se_ameba_index_chk().